webapp/packages/core-settings/src/SettingsProvider.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { ISyncExecutor, SyncExecutor } from '@cloudbeaver/core-executor';
+import { isNotNullDefined, type schema } from '@cloudbeaver/core-utils';
+
+import type { ISettingChangeData, ISettingsSource } from './ISettingsSource';
+
+export class SettingsProvider<TSchema extends schema.SomeZodObject = any> implements ISettingsSource {
+  readonly onChange: ISyncExecutor<ISettingChangeData<keyof schema.infer<TSchema>>>;
+  constructor(
+    private readonly source: ISettingsSource,
+    readonly scope: string,
+    readonly schema: TSchema,
+  ) {
+    this.onChange = new SyncExecutor();
+    source.onChange.next(
+      this.onChange,
+      data => ({ key: this.unscopedKey(data.key), value: data.value }),
+      data => data.key.startsWith(this.scope),
+    );
+  }
+
+  isReadOnly<TKey extends keyof schema.infer<TSchema>>(key: TKey): boolean {
+    return this.source.isReadOnly(this.scopedKey(key)) || false;
+  }
+
+  isEdited(key?: any): boolean {
+    return this.source.isEdited(this.scopedKey(key));
+  }
+
+  has<TKey extends keyof schema.infer<TSchema>>(key: TKey): boolean {
+    return true;
+  }
+
+  isValueDefault<TKey extends keyof schema.infer<TSchema>>(key: TKey): boolean {
+    return !this.source.has(this.scopedKey(key));
+  }
+
+  getEditedValue<TKey extends keyof schema.infer<TSchema>>(key: TKey): schema.infer<TSchema>[TKey] {
+    return this.getValue(key);
+  }
+
+  getValue<TKey extends keyof schema.infer<TSchema>>(key: TKey): schema.infer<TSchema>[TKey] {
+    let value = undefined;
+
+    if (this.source.has(this.scopedKey(key))) {
+      value = this.source.getValue(this.scopedKey(key));
+    }
+
+    const schema = this.schema.shape[key as any];
+    const result = schema.safeParse(value);
+
+    if (result.success) {
+      return result.data;
+    }
+
+    return schema.parse(undefined);
```

Comment:
```suggestion
    return schema.parse(value);
```

Reply:
parse can throw en exception, in that case when we can't parse value we want to use default value insted (schema.parse(undefined))
SettingsProvider ensures that you will always get correct setting value